packages/compose/src/hooks/use-media-query/index.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Runs a media query and returns its value when it changes.
+ *
+ * @param {string} query Media Query.
+ * @return {boolean} return value of the media query.
+ */
+export default function useMediaQuery( query ) {
```

Comment:
How would we handle these for mobile native conceptually?

Reply:
That's a good question, I think Mobile shouldn't really care about that hook. Ultimately, this hook might not be used directly by component. instead we'd use higher-level ones that can be "forked" by mobile for example `useReducedMotion` can be adapted for mobile.